### DISCUSSION TITLE
can Gnpy consider some OSP information in route selection ?

### DISCUSSION THREAD

**Discussion by 3koozy** [Telecominfraproject/oopt-gnpy#468] [General]
Hello everyone , hope your haveing a wonderful day :)
I am new to the tool , and since this feature wasn't explicitly provided in the description , I though I will ask here.

**can Gnpy consider some OSP information in route selection ?**
the scenario I am talking about is considering SRLG (Shared Risk Link Group) and weather two links in the same SRLG are sharing certain amount of the physical OSP fiber ?

this is a very common discussion in many operators , and the usual way to do this is to manually plot every service on Google map or other Geospatial tool and them calculate physical fiber sharing in KM.

**It would be lovely if we can automate this task along with route selection (to choose least fiber sharing route).**

Regards.

**Comment by 3koozy** [comment]
While reading more from the docs, I have found references to: Cable ID, and SRG in network topology definition files (ex : docs/excel.rst).

Is this information used in any shape or form as part of gnpy simulations?

**Comment by jktjkt** [comment]
Hi @3koozy, there's some support for path exclusivity [as defined by an older IETF draft](https://datatracker.ietf.org/doc/html/draft-ietf-teas-yang-path-computation-05#section-5.1). In the `synchronization/svec/disjointness` leaf, GNPy appears to implement the `node` and `link` bits from [RFC8776](https://datatracker.ietf.org/doc/html/rfc8776), but there's no support for `srlg`. Patches are welcome.
